source/pdal/pdalc_pipeline.cpp

```diff
@@ -108,26 +111,24 @@ extern "C"
         if (pipeline && buffer && size > 0)
         {
             Pipeline *ptr = reinterpret_cast<Pipeline *>(pipeline);
-            pdal::PipelineExecutor *executor = ptr->get();
             buffer[0] =  '\0';
             buffer[size - 1] = '\0';
 
-            if (executor)
+            try
             {
-                try
-                {
-                    std::string s = executor->getPipeline();
-                    std::strncpy(buffer, s.c_str(), size - 1);
-                    result = std::min(s.length(), size);
-                }
-                catch (const std::exception &e)
-                {
-                    printf("Found error while retrieving pipeline's string representation: %s\n", e.what());
-                }
-            }
+                if (! ptr->m_executed)
```

Comment:
To be honest - the error handling of this interface and the relationship of errors between the bindings and the core is kind of broken - but that is nothing new.

I do not want or have the time at the moment to do a proper job fixing that - so I would rather raise it an issue to be addressed at some point and otherwise leave the error handling very much as it was before this change. I want to get the change through as it is blocking updating the conda package to PDAL 2.5.0. 

This will produce an error message something like "Found error while retrieving pipeline's string representation: Pipeline has not been executed!" which is a reproduction of what happened with Pipeline Executor.

I have not probed into why the old Pipeline Executor did not return the string representation of the pipeline if has not been executed - I have just reproduced the experience.

Reply:
see #19